internal/hud/hud.go

```diff
@@ -92,7 +92,10 @@ func (h *Hud) Run(ctx context.Context, dispatch func(action store.Action), refre
 		h.mu.Unlock()
 	}()
 
+	h.mu.RLock()
```

Comment:
Can you elaborate on what this lock is for? `h.r` doesn't change, so i don't think it needs a lock here, though it's probably harmless

Reply:
Hi @nicks,
I know that 'h.r' will normally not change after init, but as we read this in a go routine it's the only way i find to read safely this field (tho the read lock RLock) and keep the go data racer happy.

Reply:
Ya, that makes sense, but I'm worried that this is papering over the real root cause of the bug.

When you repro this, are you running a particular test? Or running the `tilt up` tith the data race detection on?

Reply:
i run from tilt source : go run --race  ./cmd/tilt/main.go up --web-mode=prod --file=xyz/tiltdemo/Tiltfile 2>log.txt